tasks/services/works.js

```diff
@@ -36,26 +79,101 @@ function createCastOrderIntentTaskPlan (hook) {
+        query: {
+          sourceId: groupId,
+          relationshipType: 'member'
+        }
+      })
+        .then((relationships) => {
+          return Promise.all(
+            map((relationship) => {
+              const assigneeId = prop('targetId', relationship)
+              return taskPlans.create({ taskRecipeId, params, assigneeId })
+            }, relationships)
+          )
+        })
+    })
+    .then(() => hook)
+}
+
+function createCloseOrderTaskPlan (hook) {
+  const taskPlans = hook.app.service('taskPlans')
+  const taskRecipeId = taskRecipes.closeOrder.id
+  // TODO: IK: at the moment the assigneeId is automatically the user who completesOrderSetup, should be the order admin
+  const assigneeId = hook.params.agent.id
+  const completedTaskPlanId = hook.result.taskPlanId
+
+  return taskPlans.get(completedTaskPlanId)
+    .then((completedTaskPlan) => {
+      const { orderId } = completedTaskPlan.params
+      const params = {
+        // consumerAgentId: hook.data.consumerAgentId,
+        // supplierAgentId: hook.data.supplierAgentId,
+        orderId
       }
+      return taskPlans.create({ taskRecipeId, params, assigneeId })
     })
-    .then((relationships) => {
-      return Promise.all(
-        map((relationship) => {
-          const assigneeId = prop('targetId', relationship)
-          return taskPlans.create({ taskRecipeId, params, assigneeId })
-        }, relationships)
-      )
+    .then(() => {
+      return hook
+    })
+}
+
+function createCastOrderIntentTaskWorks (hook) {
+  // order has closed, get all relevant taskPlans, create a taskWork for em
+  const taskPlans = hook.app.service('taskPlans')
+  const taskWorks = hook.app.service('taskWorks')
+  const closeOrderTaskPlanId = hook.data.taskPlanId
+
+  return taskPlans.get(closeOrderTaskPlanId)
+    .then((closeOrderTaskPlan) => {
+      const orderId = closeOrderTaskPlan.params.orderId
+      // const params = { orderId, consumerAgentId: groupId, supplierAgentId }
+
+      // TODO: IK: feels like a pretty sub-optimal way of querying, probably makes sense to have orderId as it's own column
+      return taskPlans.find({
+        query: {
+          taskRecipeId: 'castIntent',
+          params: {
+            $like: `%"orderId":${orderId}%`
+          }
+        }
+      })
+        .then((taskPlans) => {
+          return Promise.all(taskPlans.map((taskPlan) => {
+            const { id, taskRecipeId, assigneeId, params } = taskPlan
+            return taskWorks.create({
+              taskPlanId: id,
+              taskRecipeId,
+              workerAgentId: assigneeId,
+              params
+            })
+          }))
+        })
+    })
+    .then(() => hook)
+}
+
+function createViewOrderSummaryTaskPlan (hook) {
+  const taskPlans = hook.app.service('taskPlans')
+  const taskRecipeId = taskRecipes.viewOrderSummary.id
+  // TODO: IK: at the moment the assigneeId is automatically the user who closes the order, should be the order admin
```

Comment:
shouldn't the order admin be the only one is assigned the "close order" task?

Reply:
fixed :)